src/main/java/pages/ProductPage.java

```diff
@@ -21,49 +20,59 @@ public ProductPage(WebDriver driver) {
         super(driver);
     }
 
-    @Override
-    public void openPage() {
-        driver.get(BASE_URL + PRODUCT_ENDPOINT);
+    public ProductPage openPage() {
+        openPage(BASE_URL + PRODUCTS_ENDPOINT);
+        return this;
     }
 
     @Override
-    public void isPageOpened() {
+    public ProductPage waitForPageOpened() {
         try {
             wait.until(ExpectedConditions.visibilityOfElementLocated(PRODUCT_LABEL_LOCATOR));
         } catch (TimeoutException e) {
             fail("Product page is not loaded. Locator " + PRODUCT_LABEL_LOCATOR + " not found");
         }
+        return this;
     }
+     public boolean isPageOpened() {
+        if (driver.findElement(PRODUCT_LABEL_LOCATOR).isDisplayed()) {
+            return true;
+        } else return false;
+     }
 
-    public void addProductToCart(String productName) {
+    public ProductPage addProductToCart(String productName) {
         driver.findElement(By.xpath(String.format(ADD_TO_CART_BUTTON_LOCATOR, productName))).click();
+        return this;
     }
 
-    public void removeProductFromCart(String productName) {
+    public ProductPage removeProductFromCart(String productName) {
         driver.findElement(By.xpath(String.format(REMOVE_BUTTON_LOCATOR, productName))).click();
+        return this;
     }
 
-    public void clickShoppingCartIcon() {
+    public CartPage clickShoppingCartIcon() {
         driver.findElement(CART_ICON_LOCATOR).click();
+        return new CartPage(driver);
     }
 
-    public void openItemPage(String productName) {
+    public ItemPage openItemPage(String productName) {
         driver.findElement(By.xpath(String.format(PRODUCT_NAME_LOCATOR, productName))).click();
+        return new ItemPage(driver);
     }
 
     public String getShoppingCartNumberFromCounter() {
         return driver.findElement(SHOPPING_CART_COUNTER).getText();
     }
 
-    public boolean addToCartBtnNameIsDisplayed(String productName) {
+    public boolean IsAddToCartBtnNameDisplayed(String productName) {
         String actualButtonName = driver.findElement(By.xpath(String.format(ADD_TO_CART_BUTTON_LOCATOR, productName))).getText();
         if (actualButtonName.contains(ADD_TO_CART_BUTTON_NAME)) {
             return true;
         }
         return false;
```

Comment:
you can simplify:
        return actualButtonName.contains(ADD_TO_CART_BUTTON_NAME);

follow the IDEA suggestions